src/main/java/org/jabref/logic/formatter/bibtexfields/RegexFormatter.java

```diff
@@ -71,7 +92,7 @@ private String replaceHonoringProtectedGroups(final String input) {
     @Override
     public String format(final String input) {
         Objects.requireNonNull(input);
-        if (regex == null) {
+        if (regex == null || replacement == null) {
```

Comment:
Could you switch to `Optional<String>` as datatype for `regex` and `replacement`? JabRef wants to get rid off `null`.

Reply:
Not easily without code-style complaints. `regex` and `replacement` are instance fields (`private final String`).

I believe the choice of using instance fields is to inherit the method `logic.cleanup.Formatter#format(String value)`, which most modifiers do, while still allowing arguments.

The behavior of most code in the citation key patterns is to either return an empty string or an unmodified string. Hence throwing an exception during construction would require code outside of `RegexFormatter` to deal with that circumstance, which I'd like to avoid due to separation of concerns.

I'd guess the alternative is to create an interface extending `logic.cleanup.Formatter` with `format(String value, String ... args)`. 

I have avoided taking these types of decisions as I am fairly new to Java and JabRef, but I can look into alternatives.